src/TwigComponent/src/Twig/TwigPreLexer.php

```diff
@@ -35,10 +35,27 @@ public function __construct(int $startingLine = 1)
 
     public function preLexComponents(string $input): string
     {
-        if (!str_contains($input, '<twig:')) {
+        // tag may be:
+        // - classical: <twig:componentName>
+        // - jsx: <ComponentName> (with a capital letter)
+
+        $isClassical = str_contains($input, '<twig:');
+        $isJsx = preg_match_all('/<([A-Z][a-zA-Z0-9_:-]+)([^>]*)>/', $input, $matches, \PREG_SET_ORDER);
+
+        if (!$isClassical && !$isJsx) {
             return $input;
         }
 
+        if ($isJsx) {
+            $componentNames = array_map(fn ($match) => $match[1], $matches);
+            $componentNames = array_unique(array_filter($componentNames));
+
+            // To simplify things in the rest of the class, we replace the component name with twig:<componentName>
+            foreach ($componentNames as $componentName) {
```

Comment:
Or maybe better, 1 loop instead of 4?  
```suggestion
            // To simplify things in the rest of the class, we replace the component name with twig:<componentName>
            foreach ($matches as $match) {
                if (!isset($componentName = $match[1])) {
                    continue;
                }
```

Reply:
I think it would be even easier in the other way.... not sure we need the 2step PATH

Reply:
For me, changing the code might actually make it slower. Your solution helps reduce iterations if all the nodes are unique.

But if a page contains the same node multiple times, it will trigger multiple `preg_replace` calls - which are much more expensive than a simple array iteration.

I think it's better to first reduce the array to a unique list, then perform the minimal number of `preg_replace` calls.

That said, happy to discuss it of course!